plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/AddFilesProcedure.java

```diff
@@ -0,0 +1,403 @@
+        Table table = catalog.loadTable(session, targetName);
+        PartitionSpec partitionSpec = table.spec();
+        Schema schema = table.schema();
+        NameMapping nameMapping = MappingUtil.create(schema);
+        Set<Integer> requiredFields = schema.columns().stream()
+                .filter(Types.NestedField::isRequired)
+                .map(Types.NestedField::fieldId)
+                .collect(toImmutableSet());
+
+        HiveStorageFormat storageFormat;
+        String sourceLocation;
+        io.trino.metastore.Table sourceTable = null;
+        SourceType sourceType;
+        if (sourceSchemaName != null && sourceTableName != null) {
+            checkProcedureArgument(location == null, "LOCATION argument must be null");
+            checkProcedureArgument(format == null, "FORMAT argument must be null");
+
+            SchemaTableName sourceName = new SchemaTableName(sourceSchemaName, sourceTableName);
+            sourceTable = metastore.getTable(sourceSchemaName, sourceTableName).orElseThrow(() -> new TableNotFoundException(sourceName));
+
+            accessControl.checkCanSelectFromColumns(null, sourceName, Stream.concat(sourceTable.getDataColumns().stream(), sourceTable.getPartitionColumns().stream())
+                    .map(Column::getName)
+                    .collect(toImmutableSet()));
+
+            checkProcedureArgument(
+                    table.schemas().size() == sourceTable.getDataColumns().size(),
+                    "Data column count mismatch: %d vs %d", table.schemas().size(), sourceTable.getDataColumns().size());
+            for (Column sourceColumn : Stream.concat(sourceTable.getDataColumns().stream(), sourceTable.getPartitionColumns().stream()).toList()) {
+                Types.NestedField targetColumn = schema.caseInsensitiveFindField(sourceColumn.getName());
+                if (targetColumn == null) {
+                    throw new TrinoException(COLUMN_NOT_FOUND, "Column '%s' does not exist".formatted(sourceColumn.getName()));
+                }
+                ColumnIdentity columnIdentity = createColumnIdentity(targetColumn);
+                org.apache.iceberg.types.Type sourceColumnType = toIcebergType(typeManager.getType(getTypeSignature(sourceColumn.getType(), DEFAULT_PRECISION)), columnIdentity);
+                if (!targetColumn.type().equals(sourceColumnType)) {
+                    throw new TrinoException(TYPE_MISMATCH, "Expected target '%s' type, but got source '%s' type".formatted(targetColumn.type(), sourceColumnType));
+                }
+            }
+
+            String transactionalProperty = sourceTable.getParameters().get(TRANSACTIONAL);
+            if (parseBoolean(transactionalProperty)) {
+                throw new TrinoException(NOT_SUPPORTED, "Adding files from transactional tables is unsupported");
+            }
+            if (!"MANAGED_TABLE".equalsIgnoreCase(sourceTable.getTableType()) && !"EXTERNAL_TABLE".equalsIgnoreCase(sourceTable.getTableType())) {
```

Comment:
*suggestion*: consider using `MANAGED_TABLE.name()` and `EXTERNAL_TABLE.name()`

Reply:
We shouldn't do that. Relates to https://github.com/trinodb/trino/commit/e0e3cf2b8917d323ed3eb65bad655bd5e019b2b0